packages/kbn-config-schema/src/types/lazy.test.ts

```diff
@@ -68,4 +68,28 @@ describe('lazy', () => {
     expect(value).toEqual(invalidSelf);
     expect(error?.message).toBe('expected value of type [string] but got [number]');
   });
+
+  it('requires a schema with a given ID to be present in the schema when validating', () => {
+    expect(() =>
+      schema
+        .object({
+          lazy: schema.lazy('unknown'),
+        })
+        .validate({ lazy: {} })
+    ).toThrow(/outside of schema boundaries/);
+  });
+
+  it('disallows duplicate ids in the same schema', () => {
```

Comment:
Decided to add a couple more unit tests to support the claims we make in the docs